Added `NODE_HEAPDUMP_FOLDER` environment variable handling

selfsk opened the pull request:
* you can specify in which folder heapdump will be saved - env variable `NODE_HEAPDUMP_FOLDER`
 * updated sigusr2 test to use NODE_HEAPDUMP_FOLDER variable

Commit:
Added `NODE_HEAPDUMP_FOLDER` handling

 * you can specify in which folder heapdump will be saved - env variable `NODE_HEAPDUMP_FOLDER`
 * updated sigusr2 test to use NODE_HEAPDUMP_FOLDER variable

Commit:
bit clarification for NODE_HEAPDUMP_FOLDER variable

bnoordhuis reviewed (commented):
Sorry for the delay. If we're adding an environment variable, then I think that instead of `NODE_HEAPDUMP_FOLDER` it could be generalized to a generic format string - let's say `NODE_HEAPDUMP_FILENAME` - with a default of `heapdump-{sec}.{usec}.heapsnapshot` and where `{sec}` and `{usec}` are replaced with their runtime values.

Commit:
Use NODE_HEAPDUMP_FILENAME variable, as template for snapshot filename

     - NODE_HEAPDUMP_FILENAME is a template for snapshot filename, include directory
     - added `replace()` function which handles replacement of {sec}, {usec} placeholders with current values
     - updated README accordingly
     - updated unit tests

selfsk commented:
@bnoordhuis thanks for feedback! I've Implemented suggested changes, please review when you get a chance. Thanks
